---
title: "Data Club"
output:
  xaringan::moon_reader:
    seal: false
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: true
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
knitr::opts_chunk$set(echo = FALSE, fig.align = 'center', dev = "svg")
```

class: title-slide center middle inverse

# High-dimensional data:<br>a different kind of big data

## Florian Privé

### Data Club - June 27, 2018

---

class: center, middle, inverse

# Introduction

---

## The data I work with: very large genotype matrices

<br>

- Each variable (column): number of mutations for **one position of the genome** (generally between 100,000 to several millions) -> **ultra-high dimensional** data

```{r, out.width="70%"}
knitr::include_graphics("https://jmtomko.files.wordpress.com/2015/12/dna-double-helix.png")
```

<br>

- Each observation (row): one individual (generally between 1000 and 1M)

.footnote[Example of a dataset I previously worked with: 15K x 280K, [celiac disease](https://doi.org/10.1038/ng.543) (~30GB)]

---

class: center, middle, inverse

# What types of analysis we do?

## (and how?)

---

## Principal Component Analysis (PCA)<br>captures population structure

<br>

```{r, out.width="70%"}
knitr::include_graphics("figures/PC-1-2.png")
```

---

### Partial PCA algorithm

<br>

- Before (for $n < 10,000$), we got the eigen decomposition from $X X^T$ (an $n \times n$ matrix). It is impossible to do it with $X^T X$ because $p$ is too large.

.footnote[n is the number of individuals and p the number of variables.]

- **Now n is also large** (both dimensions of the matrix are large), then we use algorithms based on random projections to get first PCs (usually we are interested in only first 10-20 PCs).

<br>

With my implementation, you can get first 10 PCs of a 15K x 100K matrix in one minute only.

---

<!-- class: center, middle, inverse -->

<!-- # How to analyze large genomic data? -->

<!-- --- -->

<!-- ## Our two R packages: bigstatsr and bigsnpr -->

<!-- ### Statistical tools with big matrices stored on disk -->

<!-- <br> -->

<!-- <a href="https://doi.org/10.1093/bioinformatics/bty185" target="_blank"> -->
<!-- ```{r, out.width='70%'} -->
<!-- knitr::include_graphics("figures/bty185.png") -->
<!-- ``` -->
<!-- </a> -->

<!-- <br> -->

<!-- - {bigstatsr} for many types of matrix, to be used by any field of research -->

<!-- - {bigsnpr} for functions that are specific to the analysis of genetic data -->


<!-- <br> -->

<!-- Package {bigstatsr} provides fast PCA, association and predictive models, etc. -->

<!-- --- -->

<!-- class: center, middle, inverse -->

<!-- # How to predict disease status<br>based on genotypes? -->

<!-- --- -->

<!-- ## Standard PRS - part 1: estimating effects -->

<!-- ### Genome-wide association studies (GWAS) -->

<!-- In a GWAS, each single-nucleotide polymorphism (SNP) is tested **independently**, resulting in one **effect size** $\hat\beta$ and one **p-value** $p$ for each SNP.  -->

<!-- ```{r, out.width="75%"} -->
<!-- knitr::include_graphics("figures/celiac-gwas-cut.png") -->
<!-- ``` -->

<!-- Easy combining: $PRS_i = \sum \hat\beta_j \cdot G_{i,j}$ -->

<!-- --- -->

<!-- ## Standard PRS - part 2: restricting predictors -->

<!-- ### <span style="color:#38761D">Clumping</span> + <span style="color:#1515FF">Thresholding</span> ("C+T" or just "PRS") -->

<!-- <br> -->

<!-- ```{r, out.width="90%"} -->
<!-- knitr::include_graphics("figures/GWAS2PRS.png") -->
<!-- ``` -->

<!-- $$PRS_i = \sum_{\substack{j \in S_\text{clumping} \\ p_j~<~p_T}} \hat\beta_j \cdot G_{i,j}$$ -->

<!-- --- -->

<!-- ## A more optimal approach to computing PRS? -->

<!-- In C+T: weights learned independently and heuristics for correlation and regularization. -->

<!-- #### Statistical learning -->

<!-- - joint models of all SNPs at once -->

<!-- - use regularization to account for correlated and null effects -->

<!-- - already proved useful in the litterature (Abraham et al. 2013; Okser et al. 2014; Spiliopoulou et al. 2015) -->

<!-- #### Our contribution -->

<!-- - a memory- and computation-efficient implementation to be used for biobank-scale data -->

<!-- - an automatic choice of the regularization hyper-parameter -->

<!-- - a comprehensive comparison for different disease architectures -->

<!-- --- -->

<!-- ## Comparison of methods for computing PRS -->

<!-- ### Penalized Logistic Regression -->

<!-- <br> -->

<!-- $$\arg\!\min_{\beta_0, \beta}(\lambda, \alpha)\left\{\underbrace{\frac{1}{n}\sum_{i=1}^n \log\left(1+e^{-y_i (\beta_0+x_i^T\beta)}\right)}_\text{Loss function} + \underbrace{\lambda \left((1-\alpha)\frac{1}{2}\|\beta\|_2^2 + \alpha \|\beta\|_1\right)}_\text{Penalization}\right\}$$ -->

<!-- <br> -->

<!-- *** -->

<!-- - $x$ is denoting the genotypes and covariables (e.g. principal components), -->

<!-- - $y$ is the disease status we want to predict, -->

<!-- - $\lambda$ is a regularization parameter that needs to be determined and -->

<!-- - $\alpha$ determines relative parts of the regularization $0 \le \alpha \le 1$. -->

<!-- <!-- --- --> -->

<!-- <!-- ### Efficient algorithm --> -->


<!-- <!-- - Strong rules for discarding predictors in lasso-type problems (Tibshirani et al., 2012) --> -->

<!-- <!-- - implemented in R package {biglasso} (Zeng et al., 2017) --> -->

<!-- <!-- - reimplemented in R package {bigstatsr} (Privé et al., 2018) with an early stopping criterion. --> -->

<!-- --- -->

<!-- ### Higher predictive performance with penalized logistic regression -->

<!-- <br> -->

<!-- ```{r, out.width="75%"} -->
<!-- knitr::include_graphics("figures/pres-AUC-logit.svg") -->
<!-- ``` -->

<!-- --- -->

<!-- ### Results: real Celiac phenotypes -->

<!-- ```{r, out.width="55%"} -->
<!-- knitr::include_graphics("figures/celiac-roc.svg") -->
<!-- ``` -->

<!-- ```{r, out.width="90%"} -->
<!-- knitr::include_graphics("figures/results-celiac-prs-logit.png") -->
<!-- ``` -->

<!-- --- -->

<!-- ### Results: real Celiac phenotypes -->

<!-- <br> -->

<!-- ```{r, out.width="90%"} -->
<!-- knitr::include_graphics("figures/density-scores.svg") -->
<!-- ``` -->

<!-- --- -->

<!-- class: center, middle, inverse -->

<!-- # How to combine the information<br>of multiple studies? -->

<!-- ## (possibly of different populations) -->

<!-- --- -->

<!-- ## Genetics are different between populations -->

<!-- <br> -->

<!-- ```{r, out.width="75%"} -->
<!-- knitr::include_graphics("figures/PC-1-2.png") -->
<!-- ``` -->

<!-- --- -->

<!-- ## which makes predictions fail on external populations -->

<!-- ```{r, out.width="63%", fig.cap="<small>Source: 10.1016/j.ajhg.2017.03.004</small>"} -->
<!-- knitr::include_graphics("figures/pred-pops.jpg") -->
<!-- ``` -->

<!-- --- -->

<!-- ## Genomics is failing on diversity -->

<!-- ```{r, out.width="70%", fig.cap="<small>Source: 10.1038/538161a</small>"} -->
<!-- knitr::include_graphics("figures/Genomics_ethnicity_2016_COMMENT_WEB.jpg") -->
<!-- ``` -->

<!-- --- -->

<!-- ## What can we do about it? -->

<!-- <br> -->

<!-- We can use information from other studies (possibly in other populations) -->

<!-- <br> -->

<!-- ```{r, out.width="95%"} -->
<!-- knitr::include_graphics("figures/update-base-pred.png") -->
<!-- ``` -->

<!-- <br> -->

<!-- **Will this improve prediction?** -->

<!-- <!-- .footnote[This is the topic of our next paper.] --> -->

<!-- --- -->

<!-- ## Can we learn more than just prediction? -->

<!-- <br> -->

<!-- 1. Imagine you learn a model on a large european population -->

<!-- 1. You project this predictive model on an african population in order to get a base predictor -->

<!-- 1. You learn another model on this african population to adjust from this base predictor $$y_i \sim base_i + \mu + X_i^T \beta$$ -->

<!-- <br> -->

<!--  **What can we tell about the SNPs that are used in the new model?** -->

<!-- <!-- --- --> -->

<!-- <!-- ## Future work: UK Biobank --> -->

<!-- <!-- UK Biobank is an extremely large dataset with  --> -->

<!-- <!--  - genetic data --> -->

<!-- <!--  - clinical data --> -->

<!-- <!--  - environmental data --> -->

<!-- <!-- <br> --> -->

<!-- <!-- ## Prospects --> -->

<!-- <!-- - training in one population to improve training and prediction in another population --> -->

<!-- <!-- - assess how can we combine the information provided by genetic data with clinical and environmental data, possibly in a non-linear way --> -->

---

class: center, middle, inverse

# Thanks!

<br>

Presentation available at

https://privefl.github.io/thesis-docs/data-club.html

<br>

`r icon::fa("twitter")` [privefl](https://twitter.com/privefl) &nbsp;&nbsp;&nbsp;&nbsp; `r icon::fa("github")` [privefl](https://github.com/privefl) &nbsp;&nbsp;&nbsp;&nbsp; `r icon::fa("stack-overflow")` [F. Privé](https://stackoverflow.com/users/6103040/f-priv%c3%a9)

.footnote[Slides created via R package [**xaringan**](https://github.com/yihui/xaringan).]
