---
title: "DBDS"
output:
  xaringan::moon_reader:
    seal: false
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE, width = 70)
knitr::opts_chunk$set(fig.align = 'center', dev = "svg", out.width = "70%",
                      echo = FALSE, comment = "", fig.width = 5, global.par = TRUE)
```

class: title-slide center middle inverse

<br>

# `r icon::fa_r_project()` packages bigsnpr & bigstatsr

<br>

## Statistical methods & tools for human genetics

### &mdash; particularly for prediction and ancestry inference &mdash;

<br>

### Florian Privé

---

### Motivations for developing these packages

<br>

- being able to run all my analyses within `r icon::fa_r_project()`

- frustration of having to use many different software, with different input formats, and requiring text files for parameters

- making it easier to build a chain of analyses, to perform some exploratory analyses, and to use  `r icon::fa_r_project()` packages I am already familiar with

- making it easier to develop new methods, which seemed very hard to do without using a simple matrix-like format

<br>

--

$\rightarrow$ I have been the first user of my packages, and it has made my work so much easier. I have used them for analyzing

- UK Biobank data

- iPSYCH data

---

### Data format

<br>

```{r, out.width="95%"}
knitr::include_graphics("https://raw.githubusercontent.com/privefl/R-presentation/master/memory-solution.svg")
```

---

### Overview of methods implemented

```{r, out.width="88%"}
knitr::include_graphics("figures/overview-packages2.jpg")
```

---

### References

```{css}
.small { font-size: 70% }
```


.small[
[11] Privé, Florian, et al. "Inferring disease architecture and predictive ability with LDpred2-auto." *bioRxiv* (2022): 2022-10. [[Preprint]]( https://doi.org/10.1101/2022.10.10.511629)

[10] Privé, Florian, et al. "Identifying and correcting for misspecifications in GWAS summary statistics and polygenic scores." *Human Genetics and Genomics Advances* 3.4 (2022). [[Open Access]](https://doi.org/10.1016/j.xhgg.2022.100136)

[9] Privé, Florian. "Using the UK Biobank as a global reference of worldwide populations: application to measuring ancestry diversity from GWAS summary statistics." *Bioinformatics* 38.13 (2022): 3477–3480. [[Open Access]](https://doi.org/10.1093/bioinformatics/btac348)

[8] Privé, Florian, et al. "Portability of 245 polygenic scores when derived from the UK Biobank and applied to 9 ancestry groups from the same cohort." *The American Journal of Human Genetics* 109.1 (2022): 12–23. [[Open Access]](https://doi.org/10.1016/j.ajhg.2021.11.008)

[7] Privé, Florian. "Optimal linkage disequilibrium splitting." *Bioinformatics* 38.1 (2022): 255–256. [[Open Access]](https://doi.org/10.1093/bioinformatics/btab519)

[6] Privé, Florian, et al. "LDpred2: better, faster, stronger." *Bioinformatics* 36.22–23 (2020): 5424–5431. [[Open Access]](https://doi.org/10.1093/bioinformatics/btaa1029)

[5] Privé, Florian, et al. "Efficient toolkit implementing best practices for principal component analysis of population genetic data." *Bioinformatics* 36.16 (2020): 4449–4457. [[Open Access]](https://doi.org/10.1093/bioinformatics/btaa520)

[4] Privé, Florian, et al. "Performing highly efficient genome scans for local adaptation with R package pcadapt version 4." *Molecular biology and evolution* 37.7 (2020): 2153–2154. [[Open access]](https://doi.org/10.1093/molbev/msaa053)

[3] Privé, Florian, et al. "Making the most of Clumping and Thresholding for polygenic scores." *The American Journal of Human Genetics* 105.6 (2019): 1213–1221. [[Open access]](https://doi.org/10.1016/j.ajhg.2019.11.001)

[2] Privé, Florian, et al. "Efficient implementation of penalized regression for genetic risk prediction." *Genetics* 212.1 (2019): 65–74. [[Open access]](https://doi.org/10.1534/genetics.119.302019)

[1] Privé, Florian, et al. "Efficient analysis of large-scale genome-wide data with two R packages: bigstatsr and bigsnpr." *Bioinformatics* 34.16 (2018): 2781–2787. [[Open access]](https://doi.org/10.1093/bioinformatics/bty185)

]

---

class: inverse, center, middle

# Thanks!

<br>

Presentation available at    
[bit.ly/bigsnpr_DBDS](https://bit.ly/bigsnpr_DBDS)

Learn more about the packages at    
[bit.ly/bigsnpr_extdoc](https://bit.ly/bigsnpr_extdoc)

<br>

`r icon::fa("twitter")` `r icon::fa("github")` privefl

.footnote[Slides created with R package [**xaringan**](https://github.com/yihui/xaringan)]

