<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>INSERM CSS6</title>
    <meta charset="utf-8" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link rel="shortcut icon" href="#" />
  </head>
  <body>
    <textarea id="source">




class: title-slide center middle

background-image: url("figures/logo-inserm.jpeg")
background-position: center 95%
background-repeat: no-repeat
background-size: 25%

# &amp;mdash;**MAGNIFIC**&amp;mdash;
# &lt;u&gt;Ma&lt;/u&gt;ximizing &lt;u&gt;G&lt;/u&gt;e&lt;u&gt;n&lt;/u&gt;et&lt;u&gt;i&lt;/u&gt;c &lt;u&gt;F&lt;/u&gt;indings and Pred&lt;u&gt;ic&lt;/u&gt;tion

&lt;br&gt;

## Florian Privé 

&lt;br&gt;

### CRCN INSERM&amp;ndash;CSS6 application  

&lt;br&gt;

---

&lt;style type="text/css"&gt;
.footnote2 {
  position: absolute;
  bottom: 1.6em;
  padding-right: 4em;
  font-size: 16.3px;
}
&lt;/style&gt;

### About me

&lt;br&gt;

#### Research Interests

- Statistical human genetics, especially polygenic risk scores (PRS)

- Development of statistical methods and R/C++ packages    
for efficient and powerful analyses of large-scale genetic data 

--

&lt;br&gt;

#### Professional background

- 2013–2016: Engineer in Computer Science &amp; Applied Mathematics

- 2016–2019: PhD in Computational Biology (Grenoble)

- 2019–2021: Postdoc at Aarhus University (Denmark)

- 2022–2025: Senior Researcher (promotion at the same place)

---

class: center middle inverse

# Introduction &amp; Motivation

---

### Genetic variants and GWAS

&lt;img src="figures/Overview_geneticvariants_GWAS.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### GWAS and polygenic risk scores (PRS)

&lt;br&gt;

Thanks to GWAS, we know that

- many **common** genetic variants are causal/pathogenic

- but, they usually have a **small effect** on their own (`\(\beta_j\)`)

- `\(\Rightarrow\)` a single common variant *cannot* be used as a risk factor

--

&lt;br&gt;

From GWAS data to polygenic risk scores (PRS):

- variants can be aggregated in a joint predictive model: `\(PRS = \sum_j \gamma_j G_j\)`

- `\(\Rightarrow\)` the `\(PRS\)` has a much larger effect than each `\(G_j\)`    
and *can* be used as a risk factor

---

### &lt;u&gt;Public Health&lt;/u&gt;&lt;br&gt;Using PRS to refine risk assessment from traditional risk factors

&lt;img src="figures/PRS-risk.PNG" width="85%" style="display: block; margin: auto;" /&gt;

.footnote2[
***
**In a randomized clinical trial** in the UK, a PRS added to a QRISK®2 cardiovascular disease risk calculator demonstrated robust clinical acceptance and clinical utility in the primary care setting.
]

---

### Refining breast cancer risk using a PRS (of 86 common variants)

&lt;img src="figures/PRS-cancer-risk.png" width="80%" style="display: block; margin: auto;" /&gt;

.footnote2[
***
Similar results from a study **in France**:    
Association and performance of polygenic risk scores for breast cancer among French women presenting or not a familial predisposition to the disease. *European Journal of Cancer* (2023).
]

---

class: center, middle, inverse

# A major limitation of PRS:

## their poor portability across populations

## risks exacerbating health disparities

---

### Predictive performance of PRS drops with genetic distance to training

&lt;img src="figures/ratio-dist-3.png" width="92%" style="display: block; margin: auto;" /&gt;

---

### How to explain this drop in predictive performance of PRS?

Counter-intuitive, because it has been shown that 

- **causal variants are mostly similar** across populations

- **their effect sizes are also very similar**

--

&lt;br&gt;

&lt;u&gt;The issue&lt;/u&gt;: we often don't use causal variants in practice 

&lt;img src="figures/LD-multipop.png" width="78%" style="display: block; margin: auto;" /&gt;

---

class: center, middle, inverse

# The solution:

# precisely identifying causal variants

---

### An overview of my proposed project MAGNIFIC

&lt;br&gt;

&lt;img src="figures/Overview_MAGNIFIC.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Prioritizing causal variants thanks to functional annotations (WP2)

&lt;br&gt;

&lt;img src="figures/anno-enrich2.png" width="55%" style="display: block; margin: auto;" /&gt;

---

### Prioritizing causal variants thanks to multi-ancestry data (WP3)

&lt;br&gt;

&lt;img src="figures/multiancestry-finemapping2.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### WP1 and WP4 to ensure causal variants are available

&lt;br&gt;

&lt;img src="figures/Overview_MAGNIFIC.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Feasability

- Already **developed many state-of-the-art methods** in past 9 years  

    - LDpred2, widely used for constructing polygenic scores
    
    - bigstatsr and bigsnpr, R(cpp) packages for large-scale analyses

- Published 28 papers, including **9 as first author and 2 as sole author**    
(with 2700+ citations in total, 2000+ for my 11 first-author papers)

--

- **Funding Strategy:**

  - ATIP-Avenir (applied)
  
  - ANR JCJC (planned)

--

- Already have **experience supervising** 

    - two PhD students who graduated, with **4 co-last-author papers**
    
    - ongoing: two PhD students, one research assistant, one postdoc
    
--

- Already found **several collaborators** for these work packages    
(Broad, UCLA, Helsinki, Pasteur, INRIA, etc)
    
---

### Integration into INSERM Toulouse Laboratory

&lt;br&gt;

- **Host Team**: "GenoFun: &lt;u&gt;Fun&lt;/u&gt;ctional impact of &lt;u&gt;Geno&lt;/u&gt;mic variations on disease", a Bioinformatics team at IRSD, INSERM U1220

--

- **Collaboration**:

  - **Sarah Djebali (CR INSERM)**: Expert in functional genome annotation, supporting integration of annotations
  
  - **Jean Monlong (CR INSERM)**: Specialist in pangenomes and structural variants, expanding from simply using single-nucleotide polymorphisms (SNPs)
  
  - **Other lab members**: validation of causal variants using experimental models (e.g., mice, organoids)
  
--
  
- Technical support and computational resources via **Genotoul compute cluster** (5000 cores, 83 TB RAM, 7.5 PB storage)

--

- **Collaborative Environment**: Toulouse bioinformatics, biostatistics, mathematics and informatics network (INRAE, CNRS, INSERM, Uni)

---

### Relevance to INSERM's Mission

&lt;br&gt;

My project aligns closely with INSERM's commitment to advancing public health through biomedical and bioinformatics innovation:

- **Precision Medicine**: enabling earlier and more accurate disease risk assessment

- **Reducing Health Disparities**: improving risk models for all worldwide populations

- **Bridging Research and Clinical Practice**: facilitating the integration of polygenic risk scores into clinical and public health strategies

- **Enhancing Genetic Discoveries**: Improving the identification of causal variants to identify disease mechanisms and therapeutic targets

---

class: inverse, center, middle

&lt;br&gt;

# Thank you for your attention

&lt;br&gt;
&lt;br&gt;

## Florian Privé 

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
