---
title: "INSERM CSS6"
output:
  xaringan::moon_reader:
    includes:
      in_header: "no_favicon.html"
    seal: false
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r setup, include=FALSE}
# renderthis::to_pdf("ldpred2-wcpg2023.Rmd", partial_slides = TRUE)
options(htmltools.dir.version = FALSE, width = 70)
knitr::opts_chunk$set(fig.align = 'center', dev = "svg", out.width = "70%",
                      echo = FALSE, comment = "", fig.width = 5, global.par = TRUE)
ICON_R_PROJECT <- icons::fontawesome$brands$`r-project`
ICON_TRI_EXCL  <- icons::fontawesome$solid$`exclamation-triangle`
ICON_INFO      <- icons::fontawesome$solid$`info-circle`
```

class: title-slide center middle

background-image: url("figures/logo-inserm.jpeg")
background-position: center 95%
background-repeat: no-repeat
background-size: 25%

# &mdash;**MAGNIFIC**&mdash;
# <u>Ma</u>ximizing <u>G</u>e<u>n</u>et<u>i</u>c <u>F</u>indings and Pred<u>ic</u>tion

<br>

## Florian Privé 

<br>

### CRCN INSERM&ndash;CSS6 application  

<br>

---

```{css}
.footnote2 {
  position: absolute;
  bottom: 1.6em;
  padding-right: 4em;
  font-size: 16.3px;
}
```

### About me

<br>

#### Research Interests

- Statistical human genetics, especially polygenic risk scores (PRS)

- Development of statistical methods and R/C++ packages    
for efficient and powerful analyses of large-scale genetic data 

--

<br>

#### Professional background

- 2013–2016: Engineer in Computer Science & Applied Mathematics

- 2016–2019: PhD in Computational Biology (Grenoble)

- 2019–2021: Postdoc at Aarhus University (Denmark)

- 2022–2025: Senior Researcher (promotion at the same place)

---

class: center middle inverse

# Introduction & Motivation

---

### Genetic variants and GWAS

```{r, out.width="100%"}
knitr::include_graphics("figures/Overview_geneticvariants_GWAS.png")
```

---

### GWAS and polygenic risk scores (PRS)

**Studying common diseases**, such as heart diseases, cancers, diabetes

<br>

Thanks to GWAS, we know that

- many **common** genetic variants are causal $\left(\beta_j \neq 0\right)$

- but, they usually have a **small effect size** $\beta_j$ on their own   
    
    $\Rightarrow$ <u>a common causal variant is not useful as a risk factor</u>

--

<br>

From GWAS data to polygenic risk scores (PRS):

- variants can be aggregated in a joint predictive model: $PRS = \sum_j \hat\gamma_j~G_j$

- by aggregating many small effects, the PRS can have a large effect 

    $\Rightarrow$ the $PRS$ can be useful as a risk factor

---

### <u>Public Health</u>: refining risk assessment from traditional risk factors

<u>Traditional risk factors</u>: age, smoking, pollution, low SES, diet, physical inactivity, family history, (low-frequency large-effect) genetic mutations, etc

```{r, out.width="82%"}
knitr::include_graphics("figures/PRS-risk2.png")
```

--

.footnote2[
***
PRS clinical utility in a **clinical trial**: A. Fuat et al, Eur. J. of Preventive Cardiology (2024)

Refining breast cancer genetic risk using a PRS **in France**: Y. Jiao et al, Eur. J. of Cancer (2023)
]

---

class: center, middle, inverse

# A major limitation of PRS:

## their poor portability across populations

## risks exacerbating health disparities

---

### PRS performance drops with distance from training population

```{r, out.width="98%", fig.align='left'}
knitr::include_graphics("figures/ratio-dist-4.png")
```

---

### Explanation: we often don't use causal variants in practice 

<br>

```{r, out.width="88%"}
knitr::include_graphics("figures/LD-multipop2.png")
```

---

class: center, middle, inverse

# The solution:

## identifying causal variants

## and using them in PRS

---

### Scaling methods to using 10M genetic variants (WP1)

<br>

There are ~10M common variants.

But, until recently, we did not have enough genetic diversity    
to **distinguish causal variants between highly correlated genetic variants**

- detailed biological information (functional annotations) $\rightarrow$ WP2

- data from multiple populations $\rightarrow$ WP3

$\Rightarrow$ Many analyses have been restricted to using ~1M variants

--

<br>

But **methods and data structures scale quadratically with #variants**    
(would need 100x runtime and memory to use 10M)

<br>

.center[`r icons::fontawesome("laptop")` I will optimize both (methods and data structures) `r icons::fontawesome("laptop")`]

---

### Prioritizing causal variants thanks to functional annotations (WP2)

<u>Examples of functional annotations</u>: protein-coding variants, regulatory variants, conserved variants (in regions with little evolutionary change)

**Variants in some functional categories are more likely to be causal**

```{r, out.width="95%"}
knitr::include_graphics("figures/LD-func-anno.png")
```

<br>

.center[`r icons::fontawesome("laptop")` I will integrate this information into my Bayesian PRS methodology `r icons::fontawesome("laptop")`]

---

### Prioritizing causal variants thanks to multi-ancestry data (WP3)

<br>

```{r, out.width="100%"}
knitr::include_graphics("figures/LD-multi-GWAS.png")
```

<br>

.center[`r icons::fontawesome("laptop")` I will integrate multi-ancestry data into my Bayesian PRS methodology `r icons::fontawesome("laptop")`]


---

### Overview of my proposed project MAGNIFIC

<br>

```{r, out.width="100%"}
knitr::include_graphics("figures/Overview_MAGNIFIC.png")
```

---

### Feasability

- **I have developed many efficient & powerful methods** in past 9 years  

    - LDpred2, widely used for constructing PRS + often ranked best
    
    - bigstatsr and bigsnpr, R(cpp) packages for large-scale analyses

- I have published 28 papers with 2800 citations in total,    
including **2000 for my 11 first-author papers**

--

- My **funding strategy** for building a team:

  - ATIP-Avenir (applied)
  
  - ANR JCJC (planned)

--

- I have **co-supervised several young researchers**

    - two PhD students who graduated, with **4 co-last-author papers**
    
    - ongoing: two PhD students, one research assistant, one postdoc
    
--

- I have found **several collaborators** for these work packages    
(Broad, UCLA, Helsinki, Pasteur, INRIA, etc)
    
---

### Integration into INSERM Toulouse Laboratory

<br>

- **Host Team**: "GenoFun: <u>Fun</u>ctional impact of <u>Geno</u>mic variations on disease", a Bioinformatics team at IRSD, INSERM U1220

--

- **Collaboration**:

  - **Sarah Djebali (CR INSERM)**: Expert in functional genome annotation, supporting integration of annotations
  
  - **Jean Monlong (CR INSERM)**: Specialist in pangenomes and structural variants, expanding from simply using single-nucleotide polymorphisms (SNPs)
  
  - **Other lab members**: validation of causal variants using experimental models (e.g., mice, organoids)
  
--
  
- Technical support and computational resources via **Genotoul compute cluster** (5000 cores, 83 TB RAM, 7.5 PB storage)

--

- **Collaborative Environment**: Toulouse bioinformatics, biostatistics, mathematics and informatics network (INRAE, CNRS, INSERM, Uni)

---

### Relevance to INSERM's mission

<br>

My project aligns closely with INSERM's commitment to advancing public health through biomedical and bioinformatics innovation:

- **Precision medicine**: enabling earlier and more accurate disease risk assessment

- **Reducing health disparities**: improving risk models for all worldwide populations

- **Bridging research and clinical practice**: facilitating the integration of PRS into clinical and public health strategies

- **Enhancing genetic discoveries**: improving the identification of causal variants to identify disease mechanisms and therapeutic targets

---

class: inverse, center, middle

<br>

# Thank you for your attention

<br>
<br>

## Florian Privé 

