<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Penalized methods for genetic data</title>
    <meta charset="utf-8" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
    <script src="libs/pagedtable-1.1/js/pagedtable.js"></script>
    <link href="libs/font-awesome-5.3.1/css/fontawesome-all.min.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">




class: title-slide center middle inverse

&lt;br&gt;

# Efficient penalized regression methods&lt;br&gt;for genetic prediction

&lt;br&gt;

## Florian Privé

### Prediction Modelling Presentations -- October 14, 2020

&lt;br&gt;

&lt;br&gt;

**Slides:** `https://privefl.github.io/thesis-docs/PLR-genetics.html`

---

class: inverse, center, middle

# Introduction to penalized regression models

---

## Multiple linear regression

We want to solve

`$$y = \beta_0 + \beta_1 G_1 + \cdots + \beta_p G_p + \gamma_1 COV_1 + \cdots + \gamma_q COV_q + \epsilon~.$$`
--

Let `\(\beta = (\beta_0, \beta_1, \dots, \beta_p, \gamma_1, \dots, \gamma_q)\)` and `\(X = [1; G_1; \dots;G_p; COV_1; \dots; COV_q]\)`, then

`$$y = X \beta + \epsilon~.$$`

--

This is equivalent to minimizing

`$$||y - X \beta||_2^2 =  ||\epsilon||_2^2~,$$`
--

whose solution is 

`$$\beta = (X^T X)^{-1} X^T y~.$$`

--

&lt;br&gt;

**What is the problem when analyzing genotype data?**

--

`$$n &lt; p$$`

---

## Penalization term -- `\(L_2\)` regularization

&lt;br&gt;

Instead, we can minimize

`$$||y - X \beta||_2^2 + \lambda ||\beta||_2^2~,$$`
--

whose solution is 

`$$\beta = (X^T X + \lambda I)^{-1} X^T y~.$$`

--

&lt;br&gt;

This is the L2-regularization ("**ridge**", Hoerl and Kennard, 1970); **it shrinks coefficients `\(\beta\)` towards 0**.

.footnote[https://doi.org/10.1080/00401706.1970.10488634]

---

## Penalization term -- `\(L_1\)` regularization

&lt;br&gt;

Instead, we can minimize

`$$||y - X \beta||_2^2 + \lambda ||\beta||_1~,$$`
--

which does not have any closed form but can be solved using iterative algorithms.

--

&lt;br&gt;

This is the L1-regularization ("**lasso**", Tibshirani, 1996); **it forces some of the coefficients to be equal to 0** and can be used as a means of variable selection, leading to sparse models.

.footnote[https://doi.org/10.1111/j.2517-6161.1996.tb02080.x]

---

## Penalization term -- `\(L_1\)` and `\(L_2\)` regularization

&lt;br&gt;

Instead, we can minimize

`$$||y - X \beta||_2^2 + \lambda (\alpha ||\beta||_1 + (1 - \alpha) ||\beta||_2^2)~,$$`
--

which does not have any closed form but can be solved using iterative algorithms ( `\(0 \le \alpha \le 1\)` ).

--

&lt;br&gt;

This is the L1- and L2-regularization ("**elastic-net**", Zou and Hastie, 2005); it is a compromise between the two previous penalties.

.footnote[https://doi.org/10.1111/j.1467-9868.2005.00503.x]

---

## Advantages and drawbacks of penalization

--

### Advantages

- Makes it possible to solve linear problems when `\(n &lt; p\)`

- Generally prevents overfitting (because of smaller effects)

--

### Drawback

- Add at least one hyper-parameter ( `\(\lambda\)` ) that needs to be chosen and another one if using the elastic-net regularization ( `\(\alpha\)` )

--

### Alternative

- Select a few variables before fitting the linear model (e.g. using marginal significance/p-values); heuristic: `\(p = n / 10\)`

---

## Cross-validation for choosing hyper-parameters

&lt;br&gt;

&lt;img src="https://privefl.github.io/R-presentation/crossval.png" width="70%" style="display: block; margin: auto;" /&gt;

However, if using e.g. R package {glmnet}, it can take a long time to run for genetic data.

---

class: inverse, center, middle

# A slightly different approach in {bigstatsr}

&lt;br&gt;

https://doi.org/10.1093/bioinformatics/bty185

https://doi.org/10.1534/genetics.119.302019

---

## Cross-Model Selection and Averaging (CMSA)

&lt;img src="https://raw.githubusercontent.com/privefl/paper2-PRS/master/figures/simple-CMSA.png" width="75%" style="display: block; margin: auto;" /&gt;

---

## Advantages of using {bigstasr}

&lt;br&gt;

- faster (mainly because of early-stopping criterion)

- memory efficient (because data is stored on disk)

- parallelization of fitting (easy because data on disk)

So, can be applied to huge genotype data.

---

class: inverse, center, middle

# Application to 240 phenotypes within the UK Biobank

---

## Genetic data


- **1,117,182** variants (HapMap3)

- Training based on **434,868** individuals of European ancestry +    
testing in 8636 South Asians, 1803 East Asians and 6983 Africans


```r
knitr::include_graphics("figures/ancestry.png")
```

&lt;img src="figures/ancestry.png" width="62%" style="display: block; margin: auto;" /&gt;

---

## Phenotypic data

&lt;br&gt;

-  111 continuous + 129 binary = **240 phenotypes**
    
&lt;br&gt;

Some examples:

- multiple diseases: cancers, diabetes, autoimmune, etc.

- body measures: height, BMI, BMD, etc.

- blood biochemistry: cholesterol, vitamin D, etc.

- ECG measures

- misc

---

## How fast is the LASSO implementation?

&lt;br&gt;


```r
knitr::include_graphics("../figures/timings.png")
```

&lt;img src="../figures/timings.png" width="75%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

Running time is quadratic with the number of non-zero variables.

---

## Prediction for height


---

## Limitation of polygenic scores: prediction in different ancestries

&lt;br&gt;


```r
knitr::include_graphics("figures/lasso-results.png")
```

&lt;img src="figures/lasso-results.png" width="100%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

Robust slope of pcor_other ~ pcor_eur, squared:

- 62.3% for South Asians
- 45.5% for East Asians
- 18.8% for Africans

---

class: inverse, center, middle

# Thanks!

&lt;br&gt;

Presentation available at https://privefl.github.io/thesis-docs/PLR-genetics.html.

&lt;br&gt;

<i class="fab  fa-twitter "></i> [privefl](https://twitter.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fab  fa-github "></i> [privefl](https://github.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fab  fa-stack-overflow "></i> [F. Privé](https://stackoverflow.com/users/6103040/f-priv%c3%a9)

.footnote[Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).]
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
