<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Postdoc meeting</title>
    <meta charset="utf-8" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">




class: title-slide center middle inverse

&lt;br&gt;

# Summary of my work so far

&lt;br&gt;

## Florian Priv√©

---

&lt;img src="figures/summary-table.JPG" width="100%" style="display: block; margin: auto;" /&gt;

---

&lt;br&gt;

&lt;img src="figures/paper1.JPG" width="80%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

Two R packages with

- on-disk matrix format to handle very large datasets

- conversion and integration with PLINK

- very fast PCA and GWAS implementations

- other algos

---

&lt;br&gt;

&lt;img src="figures/paper2.JPG" width="80%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

- efficient implementation of penalized linear/logistic regression for deriving polygenic scores based on individual-level data (e.g. the UK Biobank)

- show very good prediction for e.g. celiac disease and height

---

&lt;img src="figures/paper3.JPG" width="60%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

- an efficient implementation to test many parameters for Clumping and Thresholding (C+T), e.g. 

    - size of the window for clumping
    - `\(r^2\)` threshold for clumping
    - p-value threshold
    - split on other parameters such as imputation INFO score or MAF
    
- instead of choosing one model (the best), can also stack (learn a linear combination of all models) using the penalized regression implemented before

- show higher predictive performance than when tuning only the p-value threshold (and using default parameters for clumping)

---

&lt;br&gt;

&lt;img src="figures/paper4.JPG" width="80%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

- allow for directly using PLINK bed files with a few missing values

- further improved the auto algorithm for PCA (auto = e.g. removing long-range LD regions)

- show how to detect outlier individuals, or how to restrict to some homogeneous samples

- fast and unbiased projection of new individuals into some PCA space

- also talk about potential other issues in PCA (relatedness, sample size imbalance)

---

&lt;br&gt;

&lt;img src="figures/paper6.JPG" width="80%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

- new implementation of LDpred, now in {bigsnpr}

- more robust (e.g. to long-range LD regions such as HLA) and to large GWAS sample sizes `\(\Rightarrow\)` better polygenic scores

- also add an *auto* version, which directly learns the two parameters of LDpred, the SNP heritability `\(h^2\)` and the proportion of causal variants `\(p\)` (therefore not needing any validation/tuning sample)

---

&lt;br&gt;

&lt;img src="figures/paper7.JPG" width="80%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

- show how to define many ancestry groups using tools from paper on PCA

- derive polygenic scores (using both penalized regression and LDpred2-auto) for 245 phenotypes in the UK Biobank

- test them in 9 ancestry groups

- show bad predictive performance for other ancestries (especially Africa)

---

&lt;br&gt;

&lt;img src="figures/paper8.JPG" width="70%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

How to split the genome in independent LD blocks?

- instead of relying on heuristics, derive an optimal algorithm using dynamic programming

---

## Current and future work

&lt;br&gt;

- show and use regularized LD matrix to get even more robust polygenic scores from LDpred2

- proper validation of the estimation of parameters from LDpred2-auto (the LDpred2 paper focused only on prediction, not inference)

- other stuff
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
