<!DOCTYPE html>
<html>
  <head>
    <title>Predicting complex diseases: performance and robustness</title>
    <meta charset="utf-8">
    <meta name="author" content="Florian Privé, Hugues Aschard, Michael G.B. Blum" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/font-awesome.min.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Predicting complex diseases:<br>performance and robustness
### <strong>Florian Privé</strong>, Hugues Aschard, Michael G.B. Blum
### RECOMB-Genetics 2018

---




class: center, middle, inverse

# Introduction

---

## Disease architectures


&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="figures/disease-archi.gif" alt="&amp;lt;small&amp;gt;Source: 10.1126/science.338.6110.1016&amp;lt;/small&amp;gt;" width="65%" /&gt;
&lt;p class="caption"&gt;&lt;small&gt;Source: 10.1126/science.338.6110.1016&lt;/small&gt;&lt;/p&gt;
&lt;/div&gt;

.footnote[How to predict common diseases based on common variants with small effects?] 

---

## Interest in prediction: polygenic risk scores (PRS)

- Wray, Naomi R., Michael E. Goddard, and Peter M. Visscher. "**Prediction of individual genetic risk** to disease from genome-wide association studies." Genome research 17.10 (**2007**): 1520-1528.

- Evans, David M., Peter M. Visscher, and Naomi R. Wray. "Harnessing the information contained within genome-wide association studies to improve **individual prediction of complex disease risk**." Human molecular genetics 18.18 (**2009**): 3525-3531.

- Wray, Naomi R., et al. "Pitfalls of **predicting complex traits** from SNPs." Nature Reviews Genetics 14.7 (**2013**): 507.

- Dudbridge, Frank. "Power and **predictive accuracy of polygenic risk scores**." PLoS genetics 9.3 (**2013**): e1003348.

- Chatterjee, Nilanjan, Jianxin Shi, and Montserrat García-Closas. "Developing and evaluating **polygenic risk prediction** models for stratified disease prevention." Nature Reviews Genetics 17.7 (**2016**): 392.

- Martin, Alicia R., et al. "Human demographic history impacts **genetic risk prediction** across diverse populations." The American Journal of Human Genetics 100.4 (**2017**): 635-649.

---

## Genome-wide association studies (GWAS)

In case-control studies, a GWAS test each single-nucleotide polymorphism (SNP) **independently**, computing an **effect size** `\(\beta\)` and a **p-value** `\(p\)`. 

&lt;br&gt;

&lt;img src="figures/celiac-gwas-cut.png" width="70%" style="display: block; margin: auto;" /&gt;

.footnote[Then, we can **combine these effects in a polygenic risk score (PRS)** of disease.]

---

## Widely-used method to compute PRS

**GWAS + &lt;span style="color:#38761D"&gt;Clumping&lt;/span&gt; + &lt;span style="color:#1515FF"&gt;Thresholding&lt;/span&gt; ("C+T" or just "PRS")** (Chatterjee et al. 2013; Dudbridge 2013; Wray et al. 2007)

`$$PRS_i = \sum_{\substack{j \in S_\text{clumping} \\ p_j~&lt;~p_T}} \beta_j \cdot G_{i,j}$$`

&lt;img src="figures/GWAS2PRS.png" width="85%" style="display: block; margin: auto;" /&gt;

.footnote[Weights learned independently and heuristics for correlation and regularization.]

---

## A more optimal approach to computing PRS?

&lt;br&gt;

#### Statistical learning

- joint models of all SNPs at once

- use regularization to account for correlated and null effects

- already proved useful in the litterature (Abraham et al. 2013; Okser et al. 2014; Spiliopoulou et al. 2015)

&lt;br&gt;

#### Our contribution

- a memory- and computation-efficient implementation to be used for **biobank-scale data**

- an automatic choice of the regularization hyper-parameter

- a comprehensive comparison for different disease architectures

---

class: center, middle, inverse

# Methods

---

## Penalized Logistic Regression

&lt;br&gt;

&lt;Small&gt;$$\arg\!\min_{\beta_0,~\beta}(\lambda, \alpha)\left\{  \underbrace{ -\sum_{i=1}^n \left( y_i \log\left(p_i\right) + (1 - y_i) \log\left(1 - p_i\right) \right) }_\text{Loss function}   +   \underbrace{ \lambda \left((1-\alpha)\frac{1}{2}\|\beta\|_2^2 + \alpha \|\beta\|_1\right) }_\text{Penalization}  \right\}$$&lt;/Small&gt;

&lt;br&gt;

***

- `\(p_i=1/\left(1+\exp\left(-(\beta_0 + x_i^T\beta)\right)\right)\)`

- `\(x\)` is denoting the genotypes and covariables (e.g. principal components), 

- `\(y\)` is the disease status we want to predict, 

- `\(\lambda\)` is a regularization parameter that needs to be determined and

- `\(\alpha\)` determines relative parts of the regularization `\(0 \le \alpha \le 1\)`. 

---

### Efficient algorithm

- Sequential strong rules for discarding predictors in lasso-type problems (Tibshirani et al., 2012)

- implemented in R package {biglasso} (Zeng et al., 2017)

- reimplemented in our R package {bigstatsr} that uses memory-mapping to matrices stored on disk

&lt;a href="https://doi.org/10.1093/bioinformatics/bty185" target="_blank"&gt;
&lt;img src="figures/bty185.png" width="60%" style="display: block; margin: auto;" /&gt;
&lt;/a&gt;

&lt;br&gt;

&lt;br&gt;

&lt;br&gt;

Let us focus on the choice of the hyper-parameter `\(\lambda\)` and how it can make our algorithm faster.

---

### Choice of the hyper-parameter `\(\lambda\)`

&lt;img src="figures/simple-CMSA.png" width="80%" style="display: block; margin: auto;" /&gt;

---

## Comprehensive simulations: varying many parameters

#### Simulation models

&lt;img src="figures/table-simus.png" width="90%" style="display: block; margin: auto;" /&gt;

#### Methods

- PRS: 
    - PRS-all (no p-value thresholding), 
    - PRS-stringent (GWAS threshold of significance) and
    - PRS-max (best prediction for all thresholds, considered as an upper-bound)

- logit-simple: penalized logistic regression with auto `\(\lambda\)`

---

## Predictive performance measures

**AUC** (Area Under the ROC Curve) and partial AUC (FPR &lt; 10%) are used.

&lt;img src="https://i.stack.imgur.com/5x3Xj.png" width="55%" style="display: block; margin: auto;" /&gt;

`$$\text{AUC} = P(S_\text{case} &gt; S_\text{control})$$`

---

class: center, middle, inverse

# Results

---

### Higher predictive performance with logit-simple

&lt;img src="figures/pres-AUC-logit.svg" width="75%" style="display: block; margin: auto;" /&gt;

.footnote[Penalized logistic regression provides higher predictive performance in the cases that matter, especially when there are correlated variables.]

---

### Predictive performance of C+T method varies with threshold

&lt;img src="figures/pres-AUC-PRS.svg" width="75%" style="display: block; margin: auto;" /&gt;

.footnote[Recall that prediction of PRS-max is an upper-bound of the prediction provided by the C+T method.]

---

### Prediction with logit-simple is improving faster

&lt;img src="figures/pres-AUC-ntrain.svg" width="75%" style="display: block; margin: auto;" /&gt;

.footnote[Performance of methods improve with larger sample size. Yet, penalized logistic regression is improving faster than the C+T method.]

---

## Real data

&lt;br&gt;

#### Celiac disease

- intolerance to gluten

- only treatment: gluten-free diet

- heritability: 57-87% (Nisticò et al. 2006)

- prevalence: 1-6%

&lt;br&gt;

#### Case-control study for the celiac disease (Dubois et al. 2010)

- ~15,000 individuals

- ~280,000 SNPs

- ~30% cases

---

### Results: real Celiac phenotypes

&lt;img src="figures/results-celiac-prs-logit.png" width="80%" style="display: block; margin: auto;" /&gt;

&lt;img src="figures/celiac-roc.svg" width="55%" style="display: block; margin: auto;" /&gt;


---

class: center, middle, inverse

# Discussion

---

### Summary of our penalized regression as compared to the C+T method

- A more **optimal** approach for predicting complex diseases

- models that are **linear** and very **sparse** 

- very **fast**

- **automatic choice** for the regularization parameter

- can be extended to capture also recessive and dominant effects

&lt;br&gt;

### Prospects: future work with the UK Biobank

- use of external summary statistics to improve models

- generalization to external populations

- integration of clinical and environmental data

---

class: center, middle, inverse

# Thanks!

&lt;br&gt;

Presentation: https://privefl.github.io/thesis-docs/recomb18.html

R package {bigstatsr}: https://github.com/privefl/bigstatsr

R package {bigsnpr}: https://github.com/privefl/bigsnpr

&lt;br&gt;

<i class="fa  fa-twitter "></i> [privefl](https://twitter.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fa  fa-github "></i> [privefl](https://github.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fa  fa-stack-overflow "></i> [F. Privé](https://stackoverflow.com/users/6103040/f-priv%c3%a9)

.footnote[Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).]










---

## Real genotype data

Use real data from a case-control study for the Celiac disease.

&lt;img src="figures/PC1-4.png" width="95%" style="display: block; margin: auto;" /&gt;

Keep only **controls** from the **UK** and **not deviating from the robust Malahanobis distance**.

---

## Simulate new phenotypes

### The liability-threshold model

&lt;img src="figures/LTM.png" width="65%" style="display: block; margin: auto;" /&gt;

---

### Two models of liability

#### A "simple" model

`$$y_i = \underbrace{\sum_{j\in S_\text{causal}} w_j \cdot \widetilde{G_{i,j}}}_\text{genetic effect} + \underbrace{\epsilon_i}_\text{environmental effect}$$` 

#### A "fancy" model

`$$y_i = \underbrace{\sum_{j\in S_\text{causal}^{(1)}} w_j \cdot \widetilde{G_{i,j}}}_\text{linear} + \underbrace{\sum_{j\in S_\text{causal}^{(2)}} w_j \cdot \widetilde{D_{i,j}}}_\text{dominant} + \underbrace{\sum_{\substack{k=1 \\ j_1=e_k^{(3.1)} \\ j_2=e_k^{(3.2)}}}^{k=\left|S_\text{causal}^{(3.1)}\right|} w_{j_1} \cdot \widetilde{G_{i,j_1} G_{i,j_2}}}_\text{interaction} + \epsilon_i$$` 

***

- `\(w_j\)` are **weights** (generated with a Gaussian or a Laplace distribution)
- `\(G_{i,j}\)` is the **allele count** of individual `\(i\)` for SNP `\(j\)`
- `\(D_{i,j} = \mathbf{1}\left\{G_{i,j} \neq 0\right\}\)`

---

### Extension via feature engineering

&lt;br&gt;

We construct a separate dataset with, for each SNP variable, two more variables coding for recessive and dominant effects.

&lt;br&gt;

&lt;img src="figures/triple.png" width="100%" style="display: block; margin: auto;" /&gt;

.footnote[We call these two methods "logit-simple" and "logit-triple".]

---

### Feature engineering improves prediction

&lt;img src="figures/pres-triple.svg" width="90%" style="display: block; margin: auto;" /&gt;

---

### Prediction with logit-simple is improving faster

&lt;br&gt;

&lt;img src="figures/pres-AUC-chr6.svg" width="80%" style="display: block; margin: auto;" /&gt;

&lt;!-- .footnote[Aims at increasing the polygenicity of the simulated models and at virtually increasing the sample size.] --&gt;
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"slideNumberFormat": "%current% / 22"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
