---
title: "Predicting complex diseases:<br>performance and robustness"
author: "**Florian Privé**, Hugues Aschard, Michael G.B. Blum"
date: "RECOMB-Genetics 2018"
output:
  xaringan::moon_reader:
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      slideNumberFormat: "%current% / 22"
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
knitr::opts_chunk$set(fig.align = "center", dev = "svg",
                      fig.asp = 0.7, out.width = "80%", echo = FALSE)
```

class: center, middle, inverse

# Introduction

---

## Disease architectures


```{r, out.width="65%", fig.cap = "<small>Source: 10.1126/science.338.6110.1016</small>"}
knitr::include_graphics("figures/disease-archi.gif")
```

.footnote[How to predict common diseases based on common variants with small effects?] 

---

## Interest in prediction: polygenic risk scores (PRS)

- Wray, Naomi R., Michael E. Goddard, and Peter M. Visscher. "**Prediction of individual genetic risk** to disease from genome-wide association studies." Genome research 17.10 (**2007**): 1520-1528.

- Evans, David M., Peter M. Visscher, and Naomi R. Wray. "Harnessing the information contained within genome-wide association studies to improve **individual prediction of complex disease risk**." Human molecular genetics 18.18 (**2009**): 3525-3531.

- Wray, Naomi R., et al. "Pitfalls of **predicting complex traits** from SNPs." Nature Reviews Genetics 14.7 (**2013**): 507.

- Dudbridge, Frank. "Power and **predictive accuracy of polygenic risk scores**." PLoS genetics 9.3 (**2013**): e1003348.

- Chatterjee, Nilanjan, Jianxin Shi, and Montserrat García-Closas. "Developing and evaluating **polygenic risk prediction** models for stratified disease prevention." Nature Reviews Genetics 17.7 (**2016**): 392.

- Martin, Alicia R., et al. "Human demographic history impacts **genetic risk prediction** across diverse populations." The American Journal of Human Genetics 100.4 (**2017**): 635-649.

---

## Genome-wide association studies (GWAS)

In case-control studies, a GWAS test each single-nucleotide polymorphism (SNP) **independently**, computing an **effect size** $\beta$ and a **p-value** $p$. 

<br>

```{r, out.width="70%"}
knitr::include_graphics("figures/celiac-gwas-cut.png")
```

.footnote[Then, we can **combine these effects in a polygenic risk score (PRS)** of disease.]

---

## Widely-used method to compute PRS

**GWAS + <span style="color:#38761D">Clumping</span> + <span style="color:#1515FF">Thresholding</span> ("C+T" or just "PRS")** (Chatterjee et al. 2013; Dudbridge 2013; Wray et al. 2007)

$$PRS_i = \sum_{\substack{j \in S_\text{clumping} \\ p_j~<~p_T}} \beta_j \cdot G_{i,j}$$

```{r, out.width="85%"}
knitr::include_graphics("figures/GWAS2PRS.png")
```

.footnote[Weights learned independently and heuristics for correlation and regularization.]

---

## A more optimal approach to computing PRS?

<br>

#### Statistical learning

- joint models of all SNPs at once

- use regularization to account for correlated and null effects

- already proved useful in the litterature (Abraham et al. 2013; Okser et al. 2014; Spiliopoulou et al. 2015)

<br>

#### Our contribution

- a memory- and computation-efficient implementation to be used for **biobank-scale data**

- an automatic choice of the regularization hyper-parameter

- a comprehensive comparison for diffirent disease architectures

---

class: center, middle, inverse

# Methods

---

## Penalized Logistic Regression

<br>

<Small>$$\arg\!\min_{\beta_0,~\beta}(\lambda, \alpha)\left\{  \underbrace{ -\sum_{i=1}^n \left( y_i \log\left(p_i\right) + (1 - y_i) \log\left(1 - p_i\right) \right) }_\text{Loss function}   +   \underbrace{ \lambda \left((1-\alpha)\frac{1}{2}\|\beta\|_2^2 + \alpha \|\beta\|_1\right) }_\text{Penalization}  \right\}$$</Small>

<br>

***

- $p_i=1/\left(1+\exp\left(-(\beta_0 + x_i^T\beta)\right)\right)$

- $x$ is denoting the genotypes and covariables (e.g. principal components), 

- $y$ is the disease status we want to predict, 

- $\lambda$ is a regularization parameter that needs to be determined and

- $\alpha$ determines relative parts of the regularization $0 \le \alpha \le 1$. 

---

### Efficient algorithm

- Sequential strong rules for discarding predictors in lasso-type problems (Tibshirani et al., 2012)

- implemented in R package {biglasso} (Zeng et al., 2017)

- reimplemented in R package {bigstatsr} (Privé et al., 2018) with Cross-Model Selection and Averaging (CMSA)

<br>

<a href="https://doi.org/10.1093/bioinformatics/bty185" target="_blank">
```{r, out.width="65%"}
knitr::include_graphics("figures/bty185.png")
```
</a>

.footnote[Package {bigstatsr} and {bigsnpr} use memory-mapping to matrices stored on risk to handle biobank-scale data.]
---

### Cross-Model Selection and Averaging (CMSA)

#### Automatic choice of the regularization hyper-parameter, in turn

<br>

```{r, out.width="95%"}
knitr::include_graphics("figures/CMSA-explained.svg")
```

- dashed line: $\lambda$ corresponding to model used
- dotted line: early stopping $\lambda$

---

## Comprehensive simulations: varying many parameters

```{r, out.width="95%"}
knitr::include_graphics("figures/table2-scenarios.png")
```

#### Models

- simple: only additive effects
- fancy: additive, dominant and interaction-type effects

#### Methods

- PRS: PRS-all (no p-value thresholding), PRS-stringent (GWAS threshold of significance) and PRS-max (best prediction for all thresholds, considered as an upper-bound)

- logit-simple: penalized logistic regression with $\alpha = 0.5$ and CMSA

---

## Predictive performance measures

**AUC** (Area Under the ROC Curve) is used.

```{r, out.width="45%"}
knitr::include_graphics("https://i.stack.imgur.com/5x3Xj.png")
```

$$\text{AUC} = P(S_\text{case} > S_\text{control})$$

.footnote[As a second measure, the **partial AUC** for specificities between 90% and 100% is also reported.]

---

class: center, middle, inverse

# Results

---

### Higher predictive performance with logit-simple

```{r, out.width="75%"}
knitr::include_graphics("figures/pres-AUC-logit.svg")
```

.footnote[Penalized logistic regression provides higher predictive performance in the cases that matter, especially when there are correlated variables.]

---

### Predictive performance of C+T method varies with threshold

```{r, out.width="75%"}
knitr::include_graphics("figures/pres-AUC-PRS.svg")
```

.footnote[Recall that prediction of PRS-max is an upper-bound of the prediction provided by the C+T method.]

---

### Prediction with logit-simple is improving faster

```{r, out.width="75%"}
knitr::include_graphics("figures/pres-AUC-ntrain.svg")
```

.footnote[Performance of methods improve with larger sample size. Yet, penalized logistic regression is improving faster than the C+T method.]

---

## Real data

<br>

#### Celiac disease

- intolerance to gluten

- only treatment: gluten-free diet

- heritability: 57-87% (Nisticò et al. 2006)

- prevalence: 1-6%

<br>

#### Case-control study for the celiac disease (Dubois et al. 2010)

- ~15,000 individuals

- ~280,000 SNPs

- ~30% cases

---

### Results: real Celiac phenotypes

```{r, out.width="55%"}
knitr::include_graphics("figures/celiac-roc.svg")
```

```{r, out.width="80%"}
knitr::include_graphics("figures/results-celiac.png")
```

---

class: center, middle, inverse

# Discussion

---

### Summary of our penalized regression as compared to the C+T method

- A more **optimal** approach for predicting complex diseases

- models that are **linear** and very **sparse** 

- very **fast**

- **automatic choice** for the regularization parameter

- can be extended to capture also recessive and dominant effects

<br>

### Prospects: future work with the UK Biobank

- use of external summary statistics to improve models

- generalization to external populations

- integration of clinical and environmental data

---

class: center, middle, inverse

# Thanks!

<br>

Presentation: https://privefl.github.io/thesis-docs/recomb18.html

R package {bigstatsr}: https://github.com/privefl/bigstatsr

R package {bigsnpr}: https://github.com/privefl/bigsnpr

<br>

`r icon::fa("twitter")` [privefl](https://twitter.com/privefl) &nbsp;&nbsp;&nbsp;&nbsp; `r icon::fa("github")` [privefl](https://github.com/privefl) &nbsp;&nbsp;&nbsp;&nbsp; `r icon::fa("stack-overflow")` [F. Privé](https://stackoverflow.com/users/6103040/f-priv%c3%a9)

.footnote[Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).]










---

## Real genotype data

Use real data from a case-control study for the Celiac disease.

```{r, out.width="95%"}
knitr::include_graphics("figures/PC1-4.png")
```

Keep only **controls** from the **UK** and **not deviating from the robust Malahanobis distance**.

---

## Simulate new phenotypes

### The liability-threshold model

```{r, out.width="65%"}
knitr::include_graphics("figures/LTM.png")
```

---

### Two models of liability

#### A "simple" model

$$y_i = \underbrace{\sum_{j\in S_\text{causal}} w_j \cdot \widetilde{G_{i,j}}}_\text{genetic effect} + \underbrace{\epsilon_i}_\text{environmental effect}$$ 

#### A "fancy" model

$$y_i = \underbrace{\sum_{j\in S_\text{causal}^{(1)}} w_j \cdot \widetilde{G_{i,j}}}_\text{linear} + \underbrace{\sum_{j\in S_\text{causal}^{(2)}} w_j \cdot \widetilde{D_{i,j}}}_\text{dominant} + \underbrace{\sum_{\substack{k=1 \\ j_1=e_k^{(3.1)} \\ j_2=e_k^{(3.2)}}}^{k=\left|S_\text{causal}^{(3.1)}\right|} w_{j_1} \cdot \widetilde{G_{i,j_1} G_{i,j_2}}}_\text{interaction} + \epsilon_i$$ 

***

- $w_j$ are **weights** (generated with a Gaussian or a Laplace distribution)
- $G_{i,j}$ is the **allele count** of individual $i$ for SNP $j$
- $D_{i,j} = \mathbf{1}\left\{G_{i,j} \neq 0\right\}$

---

### Extension via feature engineering

<br>

We construct a separate dataset with, for each SNP variable, two more variables coding for recessive and dominant effects.

<br>

```{r, out.width="100%"}
knitr::include_graphics("figures/triple.png")
```

.footnote[We call these two methods "logit-simple" and "logit-triple".]

---

### Feature engineering improves prediction

```{r, out.width="90%"}
knitr::include_graphics("figures/pres-triple.svg")
```

---

### Prediction with logit-simple is improving faster

<br>

```{r}
knitr::include_graphics("figures/pres-AUC-chr6.svg")
```

<!-- .footnote[Aims at increasing the polygenicity of the simulated models and at virtually increasing the sample size.] -->
