<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>19_Sep_2024</title>
    <meta charset="utf-8" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link rel="shortcut icon" href="#" />
  </head>
  <body>
    <textarea id="source">




class: title-slide center middle inverse

&lt;br&gt;

# Principal Component Analysis (PCA) and Genetic Ancestry Inference (GAI)

&lt;br&gt;

## LBBE Seminar (19/09/24)

&lt;br&gt;

## Florian Priv√© 
### Aarhus University (DK)
#### &lt;svg viewBox="0 0 512 512" style="height:1em;position:relative;display:inline-block;top:.1em;fill:white;" xmlns="http://www.w3.org/2000/svg"&gt;  &lt;path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"&gt;&lt;/path&gt;&lt;/svg&gt; &lt;svg viewBox="0 0 496 512" style="height:1em;position:relative;display:inline-block;top:.1em;fill:white;" xmlns="http://www.w3.org/2000/svg"&gt;  &lt;path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"&gt;&lt;/path&gt;&lt;/svg&gt; privefl

---

class: inverse, center, middle

# Introduction

---

### About me

&lt;br&gt;

--

- Engineer in Informatics and Applied Mathematics (ENSIMAG, 2013&amp;#8211;2016)

--

- PhD (2016&amp;#8211;2019) with Michael Blum (TIMC, Grenoble) and Hugues Aschard (Institut Pasteur) in computational biology (statistical genetics),    
working mostly on polygenic scores and PCA 

--

- Founder and organizer of the R user group of Grenoble (2017&amp;#8211;2019)

--

- Teaching an advanced R course since 2018

--

- Postdoc (2019&amp;#8211;2022) and Senior Researcher (2022&amp;#8211;) at Aarhus University (Denmark)

--

- Working remotely from France since December 2020    
(not too far from here)

---

### The data I work with: very large genotype matrices

&lt;br&gt;

&lt;img src="https://jmtomko.files.wordpress.com/2015/12/dna-double-helix.png" width="90%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

--

- Each variable (column): number of mutations for **one position of the genome** (generally between 100,000 to several millions are available)    
-&gt; **ultra-high dimensional** data

--

- Each observation (row): one individual (generally between 1000 and 1M)

.footnote[Example of a dataset I work with: 500K x 90M, the UK Biobank]

---

### Polygenic Risk Scores (PRS) for personalized medicine

#### Application: to identify high risk individuals

&lt;img src="figures/PRS.png" width="90%" style="display: block; margin: auto;" /&gt;

.footnote[My main focus: developing PRS methods.]

---

class: inverse, center, middle

# Principal Component Analysis (PCA)
# and Population Structure

&lt;br&gt;

#### (my side projects)

---

### PCA of genetic data captures continental population structure

&lt;br&gt;

&lt;img src="figures/PCA-UKBB-Bycroft.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### PCA also captures sub-continental population structure

&lt;img src="figures/PCA-POPRES-EUR.png" width="80%" style="display: block; margin: auto;" /&gt;

---

### Distance in PCA measures genetic distance

&lt;br&gt;

&lt;img src="figures/compare-Euclidean-to-Fst.png" width="100%" style="display: block; margin: auto;" /&gt;

---

class: inverse, center, middle

# PCA best practices and pitfalls

---

&lt;img src="figures/paper4-2.png" width="100%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

--

- Outlier sample detection &amp; relatedness filtering

- Imbalance of ancestry groups

- Projection of individuals onto the PCA space

- Automatic detection and removal of LD    
(LD: Linkage Disequilibrium `\(\rightarrow\)` correlation between genetic variants)

--

&lt;br&gt;

I provide fast &amp; efficient methods to solve these problems.


---

### Individual outlier detection

&lt;br&gt;


``` r
prob &lt;- bigutilsr::prob_dist(obj.svd$u, ncores = nb_cores())
S &lt;- prob$dist.self / sqrt(prob$dist.nn)
```

&lt;br&gt;

&lt;img src="figures/PCA-outliers.jpeg" width="100%" style="display: block; margin: auto;" /&gt;

---

### PCA projection

#### Beware of shrinkage when n &lt; p

`\(X = U \Delta V^T\)` (PC scores: `\(U \Delta\)`; PC loadings: `\(V\)`; simple projection (red): `\(X_\text{new} V\)`)

&lt;br&gt;

&lt;img src="figures/pca-proj.jpeg" width="100%" style="display: block; margin: auto;" /&gt;

---

### Check whether PCA captures LD (instead of population structure)

&lt;br&gt;

&lt;img src="figures/loadings.png" width="100%" style="display: block; margin: auto;" /&gt;

---

class: inverse, center, middle

# Genetic Ancestry Deconvolution

## (with reference populations)

---

### Summix: ancestry estimation from GWAS allele frequencies (AF)

Source: 10.1016/j.ajhg.2021.05.016

&lt;br&gt;

Estimate ancestry proportions `\(\alpha\)` such that

&lt;style type="text/css"&gt;
.small2 {
  font-size: 17px;
}
&lt;/style&gt;

&lt;span class="small2"&gt; `$$AF = \alpha_\text{AFR} AF_\text{AFR} + \alpha_\text{EAS} AF_\text{EAS} + \alpha_\text{EUR} AF_\text{EUR} + \alpha_\text{SAS} AF_\text{SAS} + \alpha_\text{IAM} AF_\text{IAM} + \epsilon ~,$$` &lt;/span&gt;

where all `\(\alpha\)` are positive and sum to 1.

--

&lt;br&gt;

More general formulation:

&lt;span class="small2"&gt; `$$\min_{\alpha_l \ge 0 \\ \sum_{\alpha_l}=1} ~~ \sum_{j=1}^M \left( AF_j^{(0)} - \sum_{l=1}^L \alpha_l AF_j^{(l)} \right)^2$$` &lt;/span&gt;

---

&lt;img src="figures/paper9-2.png" width="100%" style="display: block; margin: auto;" /&gt;

--

&lt;br&gt;

- Curate the UK Biobank to define 18 worldwide reference groups

--

- Use PCA to maximize power to distinguish between these populations:

    &lt;span class="small2"&gt; `$$\min_{\alpha_l \ge 0 \\ \sum_{\alpha_l}=1} ~~ \sum_{j=1}^M \left( AF_j^{(0)} - \sum_{l=1}^L \alpha_l AF_j^{(l)} \right)^2$$` &lt;/span&gt;
    
    is replaced by
    
    &lt;span class="small2"&gt; `$$\min_{\alpha_l \ge 0 \\ \sum_{\alpha_l}=1} ~~ \sum_{k=1}^K \left( PC_k^{(0)} - \sum_{l=1}^L \alpha_l PC_k^{(l)} \right)^2$$` &lt;/span&gt;
    
---

&lt;img src="figures/ancestry-bigsnpr.png" width="100%" style="display: block; margin: auto;" /&gt;

--

&lt;br&gt;

&lt;img src="figures/ancestry-summix.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Admixture coefficients for individual-level data

&lt;br&gt;

**My developed method also works for individual-level data!**

(by simply using genotypes, divided by 2, in place of allele frequencies)

This is similar to the projection analysis from ADMIXTURE,    
but should have more power..

--

&lt;br&gt;

Application to iPSYCH (genetic study in Denmark):

Out of 134K individuals, can identify many non-European individuals:

- Middle East: 2600
- East Africa: 450
- North Africa: 330
- South Asia: 840
- East Asia: 280

---

class: inverse, center, middle

# Genetic Ancestry Deconvolution

## (without reference populations)

---

### Study population: the Simons Genome Diversity Project

345 individuals from 142 worldwide populations

&lt;img src="figures/sgdp_map.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Using previous reference populations and PCA projection

&lt;img src="figures/sgdp_proj.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Admixture coefficients (by country)

&lt;img src="figures/sgdp_anc1.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Admixture coefficients (by GPS coordinates)

&lt;img src="figures/sgdp_anc2.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### PCA recomputed

&lt;br&gt;

&lt;img src="figures/sgdp_pca.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Fuzzy k-means clustering (reference-free) (m = 5)

&lt;br&gt;

&lt;img src="figures/clust_FKM_5.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Fuzzy k-means clustering (reference-free) (m = 2)

&lt;br&gt;

&lt;img src="figures/clust_FKM_2.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Fuzzy k-means clustering (reference-free) (m = 1.5)

&lt;br&gt;

&lt;img src="figures/clust_FKM_1.5.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Fuzzy k-means clustering (reference-free) (m = 1.1)

&lt;br&gt;

&lt;img src="figures/clust_FKM_1.1.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Hierarchical clustering to get reference populations

&lt;br&gt;

&lt;img src="figures/clust_hclust_11.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### ADMIXTURE-like analysis from R package LEA (L = 11)

(directly on genotype data, not PCs)

&lt;img src="figures/clust_admixture_11.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### ADMIXTURE-like analysis from R package LEA (L = 12)

(directly on genotype data, not PCs)

&lt;img src="figures/clust_admixture_12.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One proposed method

&lt;br&gt;

- Iterate between 

    - estimating ancestry coefficients (for each indiv) with
    &lt;span class="small2"&gt; `$$\min_{\alpha_l \ge 0 \\ \sum_{\alpha_l}=1} ~~ \sum_{k=1}^K \left( PC_k^{(0)} - \sum_{l=1}^L \alpha_l PC_k^{(l)} \right)^2$$` &lt;/span&gt;
    
    - estimating reference positions with
    &lt;span class="small2"&gt; `$$c_l = \dfrac{\sum_x {\alpha_l(x)}^m \cdot x}{\sum_x {\alpha_l(x)}^m} ~,$$` &lt;/span&gt;
    where `\(x\)` are the PCs for one indiv, and `\(c_l = \left(PC_1^{(l)},~\dots,~PC_K^{(l)}\right)\)`
  
--

- Instead of picking L initial `\(c_l\)` at random,    
sequentially add one more PC and one more reference    
(with good starting values)

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_2.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_3.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_4.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_5.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_6.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_7.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_8.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_9.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_10.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 1.5\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_11.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### One prototyped method (with added fuzziness `\(m = 3\)`)

&lt;br&gt;

&lt;img src="figures/clust_myalgo_morefuzz.png" width="100%" style="display: block; margin: auto;" /&gt;

---

### Corresponding admixture coefficients (by GPS coordinates)

&lt;br&gt;

&lt;img src="figures/sgdp_anc3.png" width="100%" style="display: block; margin: auto;" /&gt;

---

class: inverse, center, middle

## Thank you for your attention

&lt;br&gt;

#### Presentation available at [bit.ly/privefl190924](https://bit.ly/privefl190924)

&lt;br&gt;
&lt;br&gt;

&lt;svg viewBox="0 0 512 512" style="height:1em;position:relative;display:inline-block;top:.1em;fill:white;" xmlns="http://www.w3.org/2000/svg"&gt;  &lt;path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"&gt;&lt;/path&gt;&lt;/svg&gt; &lt;svg viewBox="0 0 496 512" style="height:1em;position:relative;display:inline-block;top:.1em;fill:white;" xmlns="http://www.w3.org/2000/svg"&gt;  &lt;path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"&gt;&lt;/path&gt;&lt;/svg&gt; privefl
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
