<!DOCTYPE html>
<html>
  <head>
    <title>Polygenic Risk Scores based on statistical learning</title>
    <meta charset="utf-8">
    <meta name="author" content="Florian Privé" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/fontawesome-all.min.css" rel="stylesheet" />
    <link rel="stylesheet" href="my-theme.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Polygenic Risk Scores<br>based on statistical learning
## Thesis follow-up n°2
### Florian Privé
### October 1, 2018

---




## Outline

&lt;br&gt;

### Introduction 

### Recall of 1st year

### Results of 2nd year

### Planning for 3rd year

---

class: center, middle, inverse

# Introduction

---

## Disease architectures

&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="figures/disease-archi.gif" alt="&amp;lt;small&amp;gt;Source: 10.1126/science.338.6110.1016&amp;lt;/small&amp;gt;" width="75%" /&gt;
&lt;p class="caption"&gt;&lt;small&gt;Source: 10.1126/science.338.6110.1016&lt;/small&gt;&lt;/p&gt;
&lt;/div&gt;

---

## Disease architectures

&lt;img src="figures/disease-archi.png" width="76%" style="display: block; margin: auto;" /&gt;

.footnote2[How to derive a genetic risk score for common diseases based on common variants with small effects?] 

---

## Interest in prediction: polygenic risk scores (PRS)

- Wray, Naomi R., Michael E. Goddard, and Peter M. Visscher. "**Prediction of individual genetic risk** to disease from genome-wide association studies." Genome research 17.10 (**2007**): 1520-1528.

- Wray, Naomi R., et al. "Pitfalls of **predicting complex traits** from SNPs." Nature Reviews Genetics 14.7 (**2013**): 507.

- Dudbridge, Frank. "Power and **predictive accuracy of polygenic risk scores**." PLoS genetics 9.3 (**2013**): e1003348.

- Chatterjee, Nilanjan, Jianxin Shi, and Montserrat García-Closas. "Developing and evaluating **polygenic risk prediction** models for stratified disease prevention." Nature Reviews Genetics 17.7 (**2016**): 392.

- Martin, Alicia R., et al. "Human demographic history impacts **genetic risk prediction** across diverse populations." The American Journal of Human Genetics 100.4 (**2017**): 635-649.

.footnote2[Still a gap between current predictions and clinical utility.&lt;/br&gt;Need more optimal predictions + larger sample sizes.]

---

class: center, middle, inverse

# Recall of 1st year

### Developing tools to help

### researchers (me!) in their analysis

---

## What I want to be able to do

### Data analysis on large-scale genotype matrices!

- Be fast to test many ideas quickly

    - code should be fast
    - I shouldn't have to make many conversions between formats
    - easily combine multiple functions
    
- Not be restricted in my analysis
   
    - Basically use all I already know in <i class="fab  fa-r-project "></i>
    
- Work on my computer (interactively)

    - I have 64 GB of RAM and 12 cores
    - Working on a server is not as easy as on my computer

&lt;br&gt;&lt;center&gt;**Smooth and fast analysis!**

---

## Memory solution when working in <i class="fab  fa-r-project "></i>

&lt;br&gt;

&lt;img src="https://privefl.github.io/RR18/memory-solution.svg" width="80%" style="display: block; margin: auto;" /&gt;

---

## My first paper

&lt;a href="https://doi.org/10.1093/bioinformatics/bty185" target="_blank"&gt;
&lt;img src="figures/bty185.png" width="80%" style="display: block; margin: auto;" /&gt;
&lt;/a&gt;

&lt;br&gt;

### Two R packages: {bigstatsr} and {bigsnpr}

&lt;br&gt;

- {bigstatsr} for many types of matrix, to be used by any field of research

- {bigsnpr} for functions which are specific to the analysis of SNP arrays


.footnote[I've presented {bigstatsr} at 3 <i class="fab  fa-r-project "></i> conferences.]

---

## Comparative performance

### Computing partial SVD

&lt;img src="figures/benchmark-pca.png" width="80%" style="display: block; margin: auto;" /&gt;

---

class: center, middle, inverse

# Results of 2nd year

### Comparing models for computing

### Polygenic Risk Scores (PRS)

---

## Standard PRS - part 1: estimating effects

### Genome-wide association studies (GWAS)

In a GWAS, each single-nucleotide polymorphism (SNP) is tested **independently**, resulting in one **effect size** `\(\hat\beta\)` and one **p-value** `\(p\)` for each SNP. 

&lt;img src="figures/celiac-gwas-cut.png" width="75%" style="display: block; margin: auto;" /&gt;

Easy combining: `\(PRS_i = \sum \hat\beta_j \cdot G_{i,j}\)`

---

## Standard PRS - part 2: restricting predictors

### &lt;span style="color:#38761D"&gt;Clumping&lt;/span&gt; + &lt;span style="color:#1515FF"&gt;Thresholding&lt;/span&gt; (&lt;span style="color:#38761D"&gt;C&lt;/span&gt;+&lt;span style="color:#1515FF"&gt;T&lt;/span&gt;)

&lt;br&gt;

&lt;img src="figures/GWAS2PRS2.png" width="90%" style="display: block; margin: auto;" /&gt;

`$$PRS_i = \sum_{\substack{j \in S_\text{clumping} \\ p_j~&lt;~p_T}} \hat\beta_j \cdot G_{i,j}$$`

---

## A more optimal approach to computing PRS?

In C+T: weights learned independently and heuristics for correlation and regularization.

#### Statistical learning

- joint models of all SNPs at once

- use regularization to account for correlated and null effects

- already proved useful in the litterature (Abraham et al. 2013; Okser et al. 2014; Spiliopoulou et al. 2015)

#### Our contribution

- a memory- and computation-efficient implementation to be used for biobank-scale data

- an automatic choice of the regularization hyper-parameter

- a comprehensive comparison for different disease architectures

---

class: center, middle, inverse

# Methods

---

## Penalized Logistic Regression (PLR)

&lt;br&gt;

&lt;Small&gt;$$\arg\!\min_{\beta_0,~\beta}(\lambda, \alpha)\left\{  \underbrace{ -\sum_{i=1}^n \left( y_i \log\left(p_i\right) + (1 - y_i) \log\left(1 - p_i\right) \right) }_\text{Loss function}   +   \underbrace{ \lambda \left((1-\alpha)\frac{1}{2}\|\beta\|_2^2 + \alpha \|\beta\|_1\right) }_\text{Penalization}  \right\}$$&lt;/Small&gt;

&lt;br&gt;

***

- `\(p_i=1/\left(1+\exp\left(-(\beta_0 + x_i^T\beta)\right)\right)\)`

- `\(x\)` is denoting the genotypes and covariables (e.g. principal components), 

- `\(y\)` is the disease status we want to predict, 

- `\(\lambda\)` is a regularization parameter that needs to be determined and

- `\(\alpha\)` determines relative parts of the regularization `\(0 \le \alpha \le 1\)`. 

---

### Efficient algorithm

- sequential strong rules for discarding predictors in lasso-type problems (Tibshirani et al. 2012; Zeng et al. 2017)

- implemented in our R package {bigstatsr} 

&lt;a href="https://doi.org/10.1093/bioinformatics/bty185" target="_blank"&gt;
&lt;img src="figures/bty185.png" width="65%" style="display: block; margin: auto;" /&gt;
&lt;/a&gt;

&lt;br&gt;

#### Our implementation

- uses memory-mapping to matrices stored on disk

- functioning choice of the hyper-parameter `\(\lambda\)` + embedded grid-search for `\(\alpha\)`

- early stopping criterion to fasten the overall procedure

---

### Choice of the hyper-parameter `\(\lambda\)` 

&lt;img src="figures/simple-CMSA.png" width="80%" style="display: block; margin: auto;" /&gt;

---

## Comprehensive simulations: varying many parameters

#### Simulation models

&lt;img src="figures/table2-simus.png" width="80%" style="display: block; margin: auto;" /&gt;

#### Methods

- C+T: 
    - C+T-all (no p-value thresholding), 
    - C+T-stringent (GWAS threshold of significance) and
    - C+T-max (best prediction for all thresholds, considered as an upper-bound)

- PLR: penalized logistic regression with automatic selection of hyper-parameters

---

## Predictive performance measures

**AUC** (Area Under the ROC Curve) and partial AUC (FPR &lt; 10%) are used.

&lt;img src="https://i.stack.imgur.com/5x3Xj.png" width="50%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

`$$\text{AUC} = P(S_\text{case} &gt; S_\text{control})$$`

---

class: center, middle, inverse

# Results

---

### Higher predictive performance with PLR

&lt;img src="figures/pres2-AUC-logit.svg" width="75%" style="display: block; margin: auto;" /&gt;

.footnote2[Penalized logistic regression consistently provides higher predictive performance, especially when there are correlated variables.]

---

### Predictive performance of C+T method varies with threshold

&lt;img src="figures/pres2-AUC-PRS.svg" width="75%" style="display: block; margin: auto;" /&gt;

.footnote2[Recall that prediction of C+T-max is an upper-bound of the prediction provided by the C+T method.]

---

### Prediction with PLR is improving faster (Scenario #3)

&lt;img src="figures/pres2-AUC-ntrain.svg" width="75%" style="display: block; margin: auto;" /&gt;

.footnote2[Performance of methods improve with larger sample size. Yet, PLR is improving faster than C+T.]

---

## Real data

&lt;br&gt;

#### Celiac disease

- intolerance to gluten

- only treatment: gluten-free diet

- heritability: 57-87% (Nisticò et al. 2006)

- prevalence: 1-6%

&lt;br&gt;

#### Case-control study for the celiac disease (WTCCC, Dubois et al. 2010)

- ~15,000 individuals

- ~280,000 SNPs

- ~30% cases

---

### Results: real Celiac phenotypes

&lt;img src="figures/results-celiac2.png" width="95%" style="display: block; margin: auto;" /&gt;

&lt;img src="figures/celiac-roc2.svg" width="55%" style="display: block; margin: auto;" /&gt;

---

class: center, middle, inverse

# Discussion

---

### Summary of our penalized regression as compared to the C+T method

- A more **optimal** approach for predicting complex diseases

- models that are **linear** and very **sparse** 

- very **fast**

- **automatic choice** for the regularization parameter

- can be extended to capture also recessive and dominant effects

&lt;br&gt;

### Prospects: future work using the UK Biobank

- use of external summary statistics to improve models

- better generalization to multiple populations

- integration of clinical and environmental data

---

class: center, middle, inverse

# Planning for 3rd year

### + early results usink the UK biobank

---

class: center, middle, inverse

# Pitfalls of penalized regression

---

## Penalization is too strong when there are large effects (sex for height)

`$$\text{height} \sim \text{SNPs} + \text{PCs} + \text{sex}$$`

&lt;img src="figures/pred-with-sex.png" width="75%" style="display: block; margin: auto;" /&gt;

---

## Better prediction, but still too conservative

`$$\text{base:}~~\text{height} \sim \text{sex}$$`
`$$\text{pred:}~~\text{height} \sim \text{base} + \text{SNPs} + \text{PCs} + \text{sex}$$`

&lt;img src="figures/pred-no-adj.png" width="75%" style="display: block; margin: auto;" /&gt;

---

## Post-process with some linear regression

`$$\text{base:}~~\text{height} \sim \text{sex}$$`

`$$\text{pred:}~~\text{height} \sim \text{base} + \text{SNPs} + \text{PCs} + \text{sex}$$`
`$$\text{post_pred:}~~\text{height} \sim \text{pred} \times \text{sex}$$`

&lt;img src="figures/pred-adj.png" width="75%" style="display: block; margin: auto;" /&gt;

---

class: center, middle, inverse

# How to combine the information&lt;br&gt;of multiple studies?

## (possibly of different populations)

---

## Genetics are different between populations

&lt;img src="figures/UKB-PC-1-2.png" width="60%" style="display: block; margin: auto;" /&gt;

.footnote[Produced with our R packages (450K people from the UK biobank)]

---

## which makes predictions fail on external populations

&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="figures/pred-pops.jpg" alt="&amp;lt;small&amp;gt;Source: 10.1016/j.ajhg.2017.03.004&amp;lt;/small&amp;gt;" width="63%" /&gt;
&lt;p class="caption"&gt;&lt;small&gt;Source: 10.1016/j.ajhg.2017.03.004&lt;/small&gt;&lt;/p&gt;
&lt;/div&gt;

---

## Genomics is failing on diversity

&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="figures/Genomics_ethnicity_2016_COMMENT_WEB.jpg" alt="&amp;lt;small&amp;gt;Source: 10.1038/538161a&amp;lt;/small&amp;gt;" width="70%" /&gt;
&lt;p class="caption"&gt;&lt;small&gt;Source: 10.1038/538161a&lt;/small&gt;&lt;/p&gt;
&lt;/div&gt;

---

## What can we do about it?

&lt;br&gt;

We can use information from other studies (possibly in other populations)

&lt;br&gt;

&lt;img src="figures/update-base-pred.png" width="95%" style="display: block; margin: auto;" /&gt;

&lt;br&gt;

**Will this improve prediction?**

&lt;!-- .footnote[This is the topic of our next paper.] --&gt;

---

### Planning for 3rd year

#### End of 2018

- Publication of second paper
- Exploratory analysis for 3rd paper
- Starting thesis writing

#### Early 2019

- Teaching (Advanced R for doctoral school + statistics at ENSIMAG)
- Final analysis + writing of third paper
- Getting my DNA sequenced and analyzing it!

#### Until July 2019

- Submission of 3rd paper
- Finishing writing thesis
- International R Conference useR!2019 in Toulouse

#### The end

- Answer reviews for 3rd paper
- Thesis defense

---

class: center, middle, inverse

# Thanks!

&lt;br&gt;

Presentation: https://privefl.github.io/thesis-docs/suivi-these2.html

R package {bigstatsr}: https://github.com/privefl/bigstatsr

R package {bigsnpr}: https://github.com/privefl/bigsnpr

&lt;br&gt;

<i class="fab  fa-twitter "></i> [privefl](https://twitter.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fab  fa-github "></i> [privefl](https://github.com/privefl) &amp;nbsp;&amp;nbsp;&amp;nbsp;&amp;nbsp; <i class="fab  fa-stack-overflow "></i> [F. Privé](https://stackoverflow.com/users/6103040/f-priv%c3%a9)

.footnote[Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).]










---

## Real genotype data

Use real data from a case-control study for the Celiac disease.

&lt;img src="figures/PC1-4.png" width="95%" style="display: block; margin: auto;" /&gt;

Keep only **controls** from the **UK** and **not deviating from the robust Malahanobis distance**.

---

## Simulate new phenotypes

### The liability-threshold model

&lt;img src="figures/LTM.png" width="65%" style="display: block; margin: auto;" /&gt;

---

### Two models of liability

#### Model "ADD"

`$$y_i = \underbrace{\sum_{j\in S_\text{causal}} w_j \cdot \widetilde{G_{i,j}}}_\text{genetic effect} + \underbrace{\epsilon_i}_\text{environmental effect}$$` 

#### Model "COMP"

`$$y_i = \underbrace{\sum_{j\in S_\text{causal}^{(1)}} w_j \cdot \widetilde{G_{i,j}}}_\text{linear} + \underbrace{\sum_{j\in S_\text{causal}^{(2)}} w_j \cdot \widetilde{D_{i,j}}}_\text{dominant} + \underbrace{\sum_{\substack{k=1 \\ j_1=e_k^{(3.1)} \\ j_2=e_k^{(3.2)}}}^{k=\left|S_\text{causal}^{(3.1)}\right|} w_{j_1} \cdot \widetilde{G_{i,j_1} G_{i,j_2}}}_\text{interaction} + \epsilon_i$$` 

***

- `\(w_j\)` are **weights** (generated with a Gaussian or a Laplace distribution)
- `\(G_{i,j}\)` is the **allele count** of individual `\(i\)` for SNP `\(j\)`
- `\(D_{i,j} = \mathbf{1}\left\{G_{i,j} \neq 0\right\}\)`

---

### Extension via feature engineering

&lt;br&gt;

We construct a separate dataset with, for each SNP variable, two more variables coding for recessive and dominant effects.

&lt;br&gt;

&lt;img src="figures/triple.png" width="100%" style="display: block; margin: auto;" /&gt;

.footnote[We call these two methods "PLR" and "PLR3".]

---

### Feature engineering improves prediction

&lt;img src="figures/pres2-triple.svg" width="90%" style="display: block; margin: auto;" /&gt;

---

### Prediction with PLR is improving faster (Scenario #2)

&lt;br&gt;

&lt;img src="figures/pres2-AUC-chr6.svg" width="80%" style="display: block; margin: auto;" /&gt;

&lt;!-- .footnote[Aims at increasing the polygenicity of the simulated models and at virtually increasing the sample size.] --&gt;
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"slideNumberFormat": "%current% / 41"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
