<!DOCTYPE html>
<html>
  <head>
    <title>Suivi de thèse n°1</title>
    <meta charset="utf-8">
    <meta name="author" content="Florian Privé" />
    <meta name="date" content="2017-09-28" />
    <link href="libs/remark-css/example.css" rel="stylesheet" />
    <script src="libs/htmlwidgets/htmlwidgets.js"></script>
    <script src="libs/jquery/jquery.min.js"></script>
    <script src="libs/datatables-binding/datatables.js"></script>
    <link href="libs/dt-core/css/jquery.dataTables.min.css" rel="stylesheet" />
    <link href="libs/dt-core/css/jquery.dataTables.extra.css" rel="stylesheet" />
    <script src="libs/dt-core/js/jquery.dataTables.min.js"></script>
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Suivi de thèse n°1
### Florian Privé
### September 28, 2017

---




## Outline 

1. Main objective of the thesis

2. Data analyzed

3. R packages

4. Future work

---

class: center, middle, inverse

# Main objective

---

## Compute polygenic risk scores 
### in order to differentiate a healthy person from a diseased person

&lt;img src="figures/density-scores.jpeg" width="80%" style="display: block; margin: auto;" /&gt;

---

## Usefulness

### Precision medecine

&lt;img src="https://www.ucdmc.ucdavis.edu/precision-medicine/images/pmSlide1.jpg" width="100%" style="display: block; margin: auto;" /&gt;

.footnote[Source: https://www.ucdmc.ucdavis.edu/precision-medicine/]

---

## Data analyzed for now

### case/control cohort for the celiac disease

---

## Celiac disease

### Intolerance to gluten

&lt;img src="http://www.strettoweb.com/wp-content/uploads/2016/12/celiaci.jpg" width="60%" style="display: block; margin: auto;" /&gt;

&lt;center&gt;is the only treatment.

---

## Celiac disease

### Prevalence of 1% in western countries but..

&lt;img src="https://www.beyondceliac.org/SiteData/images/FastFacts2/413d26a2a7026920/FastFacts_2.png" width="100%" style="display: block; margin: auto;" /&gt;

.footnote[Source: https://www.beyondceliac.org/celiac-disease/facts-and-figures/]

---

## Celiac disease

### The dataset: SNP array with

&lt;/br&gt;
&lt;table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt; Population &lt;/th&gt;
   &lt;th style="text-align:center;"&gt; UK &lt;/th&gt;
   &lt;th style="text-align:center;"&gt; Finland &lt;/th&gt;
   &lt;th style="text-align:center;"&gt; Netherlands &lt;/th&gt;
   &lt;th style="text-align:center;"&gt; Italy &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; Total &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Cases &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 2586 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 647 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 803 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 497 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 4533 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Controls &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 7532 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 1829 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 846 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 543 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 10750 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; Total &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 10118 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 2476 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 1649 &lt;/td&gt;
   &lt;td style="text-align:center;"&gt; 1040 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 15283 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

&lt;/br&gt;

&lt;center&gt;over ~300K SNPs&lt;/center&gt;

&lt;/br&gt;

This data would take **32GB** if stored in RAM as a standard R matrix.

---

class: center, middle, inverse

# Polygenic Risk Scores

---

## Standard model used in the human litterature

### P+T procedure, begins with a GWAS

&lt;img src="figures/celiac-gwas-cut.png" width="90%" style="display: block; margin: auto;" /&gt;

&lt;center&gt;+ Pruning and Thresholding

---

### All steps required in the P+T procedure.

&lt;img src="figures/steps-PT.svg" width="80%" style="display: block; margin: auto;" /&gt;

---

### PCA in GWAS?

&lt;br&gt;&lt;br&gt;

&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="figures/qqplot1.png" alt="Inflated Q-Q plot when not correcting for population structure." width="70%" /&gt;
&lt;p class="caption"&gt;Inflated Q-Q plot when not correcting for population structure.&lt;/p&gt;
&lt;/div&gt;

---

### PCA in GWAS?

Principal Components to adjust for the condounding effect of population structure (Patterson,
Price, and Reich 2006).

&lt;div class="figure" style="text-align: center"&gt;
&lt;img src="figures/qqplot2.png" alt="Less inflated Q-Q plot when correcting for population structure." width="70%" /&gt;
&lt;p class="caption"&gt;Less inflated Q-Q plot when correcting for population structure.&lt;/p&gt;
&lt;/div&gt;

---

## How to perform PCA for genetic data?

### On the whole (scaled) matrix

&lt;img src="figures/PC-1-2.png" width="70%" style="display: block; margin: auto;" /&gt;

---

## How to perform PCA for genetic data?

### On the whole (scaled) matrix

&lt;img src="figures/PC-3-4.png" width="70%" style="display: block; margin: auto;" /&gt;

---

## How to perform PCA for genetic data?

### On the whole (scaled) matrix **with pruning** (Abdellaoui et al. 2013)

&lt;img src="figures/PC2-3-4.png" width="70%" style="display: block; margin: auto;" /&gt;

---

## How to perform PCA for genetic data?

### On the whole (scaled) matrix **with pruning**

&lt;img src="figures/PC3-3-4.png" width="70%" style="display: block; margin: auto;" /&gt;

---

## How to perform PCA for genetic data?

### Capture long-range LD regions of chromosomes 6 and 8

&lt;img src="figures/load3-3-4.png" width="70%" style="display: block; margin: auto;" /&gt;

---

### Long-range LD regions for the human genome (Price et al. 2008)

<div id="htmlwidget-90c4b67d9cfda8256eab" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-90c4b67d9cfda8256eab">{"x":{"filter":"none","data":[[1,2,2,2,3,3,3,5,5,5,5,6,6,6,7,8,8,8,10,11,11,12,12,20,23,23,23,23,23,23,23,23,23,23],[48060567,85941853,134382738,182882739,47500000,83500000,89000000,44500000,98000000,129000000,135500000,25500000,57000000,140000000,55193285,8000000,43000000,112000000,37000000,46000000,87500000,33000000,109521663,32000000,14150264,25650264,33150264,55133704,65133704,71633704,80080511,100580511,125602146,129102146],[52060567,100407914,137882738,189882739,50000000,87000000,97500000,50500000,100500000,132000000,138500000,33500000,64000000,142500000,66193285,12000000,50000000,115000000,43000000,57000000,90500000,40000000,112021663,34500000,16650264,28650264,35650264,60500000,67633704,77580511,86080511,103080511,128102146,131602146],["hild1","hild2","hild3","hild4","hild5","hild6","hild7","hild8","hild9","hild10","hild11","hild12","hild13","hild14","hild15","hild16","hild17","hild18","hild19","hild20","hild21","hild22","hild23","hild24","hild25","hild26","hild27","hild28","hild29","hild30","hild31","hild32","hild33","hild34"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Chr<\/th>\n      <th>Start<\/th>\n      <th>Stop<\/th>\n      <th>ID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":7,"columnDefs":[{"className":"dt-right","targets":[0,1,2]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[7,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>


.footnote[Source: https://goo.gl/wTPY7n]

---

## How to perform PCA for genetic data?

### When removing long-range LD regions and pruning

&lt;img src="figures/PC4-3-4.png" width="70%" style="display: block; margin: auto;" /&gt;

.footnote[Source: https://goo.gl/wTPY7n]

---

## Not sure that people always do it correctly

- Importance of PCA in genetic association studies: 

&lt;img src="figures/price-cite.png" width="80%" style="display: block; margin: auto;" /&gt;

- Importance of pruning in computing PCA:

&lt;img src="figures/pruning-cite.png" width="80%" style="display: block; margin: auto;" /&gt;

Maybe not make another paper about it, but certainly a vignette for the package.

---

class: center, middle, inverse

# Polygenic Risk Scores

## Another approach

---

## Recall of what we want to achieve

### Predict a phenotype: pitfalls of the P+T model

- Weigths learned independently

- Correlation taken care of heuristically (with pruning)

- Regularization taken care of heuristically (with thresholding)

### A better solution?

For example, for binary outcomes, why not using

- logistic regression 

- Support Vector Machine (SVM)

on the whole matrix (+ PCs)?

---

class: center, middle, inverse

# Big Data

### Simpler solutions are easier to implement

---

## What I want to be able to do

### Data analysis on large-scale genotype matrices!

- Be fast to test many ideas quickly

    - code should be fast
    - I shouldn't have to make many conversions
    - easily combine multiple functions
    
- Not be restricted in my analysis
   
    - Basically use all I already know in R
    
- Work on my computer

    - I have 64 GB of RAM and 12 cores
    - Working on a server is not as easy as on my computer

&lt;br&gt;&lt;center&gt;**Smooth and fast analysis!**

---

## Memory problem when working in R

&lt;br&gt;

&lt;img src="figures/memory-problem.svg" width="80%" style="display: block; margin: auto;" /&gt;

---

## Memory solution when working in R

&lt;br&gt;

&lt;img src="figures/memory-solution.svg" width="80%" style="display: block; margin: auto;" /&gt;

.footnote[I don't use **bigmemory** anymore but still something very similar.]

---

## My first paper (as a preprint)

&lt;img src="figures/mypaper.png" width="80%" style="display: block; margin: auto;" /&gt;

---

## Two R packages

### bigstatsr and bigsnpr

&lt;br&gt;

- **bigstatsr** for many types of matrix, to be used by any field of research

- **bigsnpr** for functions which are specific to the analysis of SNP arrays

&lt;br&gt;

&lt;img src="figures/gad-abraham.png" width="50%" style="display: block; margin: auto;" /&gt;

.footnote[Gad Abraham would propably be one of the reviewer of my paper.]

---

## Comparative performance

### Computing partial SVD

&lt;img src="figures/benchmark-pca.png" width="80%" style="display: block; margin: auto;" /&gt;

---

class: center, middle, inverse

# Next paper

### Comparison of methods for computing PRS

---

## Assessing predictive performance

AUC (Area Under the ROC Curve) is often used.

&lt;img src="https://i.stack.imgur.com/5x3Xj.png" width="40%" style="display: block; margin: auto;" /&gt;

***

&gt; The AUC of a classifier is equivalent to the probability that the classifier will rank a randomly chosen positive instance higher than a randomly chosen negative instance. (Fawcett, 2006)

`$$\text{AUC} = P(S_\text{case} &gt; S_\text{control})$$`

---

### PRS with P+T on Celiac 

&lt;img src="figures/AUC-PRS.png" width="75%" style="display: block; margin: auto;" /&gt;

---

### PRS with bigstatsr's regularized logistic regression on Celiac 

#### `big_spLog`

&lt;img src="figures/AUC-spLog.png" width="80%" style="display: block; margin: auto;" /&gt;

---

## Choose the number of predictors

### Cross-Model Selection and Averaging (`big_CMSA`)

1. This function separates the training set in K folds (e.g. 10).

2. In turn,

    - each fold is considered as an inner validation set and the others (K - 1) folds form an inner training set,

    - the model is trained on the inner training set and the corresponding predictions (scores) for the inner validation set are computed,

    - the vector of scores which maximizes feval is determined,

    - the vector of coefficients corresponding to the previous vector of scores is chosen.

3. The K resulting vectors of coefficients are then combined into one vector.

---

## Using CMSA for `big_spLog`

### works really well

&lt;img src="figures/AUC-spLog-CMSA.png" width="80%" style="display: block; margin: auto;" /&gt;

---

class: center, middle, inverse

# Thanks!

### Presentation available at
### https://privefl.github.io/thesis-docs/suivi-these.html

.footnote[Slides created via the R package [**xaringan**](https://github.com/yihui/xaringan).]
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {window.dispatchEvent(new Event('resize'));});
(function() {var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler"); if (!r) return; s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }"; d.head.appendChild(s);})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
